kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateModelMojo.java

```diff
@@ -200,6 +201,21 @@ private void generateModel() throws MojoExecutionException, MojoFailureException
                 throw new MojoExecutionException("Unable to write file", e);
             }
 
+            // copy META-INF "generated-class-names" file
+            try {
+                final String genClassNamesPathStr = CanonicalKieModule.getGeneratedClassNamesFile(kieModule.getReleaseId());
+                final MemoryFile genClassNamesMemoryFile = (MemoryFile) mfs.getFile(genClassNamesPathStr);
+                final String genClassNamesMemoryFolderPath = genClassNamesMemoryFile.getFolder().getPath().toPortableString();
+                final Path genClassNamesDestinationPath = Paths.get(targetDirectory.getPath(), "classes", genClassNamesMemoryFolderPath, genClassNamesMemoryFile.getName());
+
+                if (!Files.exists(genClassNamesDestinationPath)) {
```

Comment:
@tkobayas 
Hi. At first sight, If I'm not wrong, this method is going to create a directory with "dot" in the name. I could also not understand why the code check for _genClassNamesDestinationPath_ but then creates _genClassNamesDestinationPath.getParent()_ - again I can't read all the code, so maybe I'm misunderstanding something

Reply:
@gitgabrio  Thank you for reviewing.

> this method is going to create a directory with "dot" in the name

Yes. the path for such metadata is `META-INF/kie/<groupId>/<artifactId>/<filename>` so `<groupId>` may contain dots. I'm not fully sure if it's a bad practice. Do you think we should replace a dot with something?

>  I could also not understand why the code check for genClassNamesDestinationPath but then creates genClassNamesDestinationPath.getParent()

Ahh, thanks. I'll fix it.

Reply:
Hi @tkobayas . I think it would be better to not have dots on directory/file names, even if now the OSes can manage it.
Some potential issues could be
1) splits on dot when managing directory/file names is a pretty common practice
2) in java, the "dotted" name is used to indicate a package: inside IDE it is impossible to differentiate if a "dotted" name is a packagename (that it is actually a series of slashed directories) or it is actually a single directory : e.g. the package "dot.something.package" is actually "dot/slash/something")
I'm pretty sure there could be other situations that does not come to my mind right now.

It may be that in the current case this may never happen but, hey, who can tell ?

Reply:
@gitgabrio  Thanks. Yeah, I agree with your point. But the path `META-INF/kie/<groupId>/<artifactId>/<filename>` has been used for other metadata files so we should fix it first before my classloader PR.

https://github.com/kiegroup/drools/blob/master/drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/CanonicalKieModule.java#L189-L195

Hi @lucamolteni  , do you agree to fix the metadata path? (= changing from a dot to standard package structure).

Reply:
TBH I don't remember why I put that structure. I remember we wanted to avoid collisions, but not the details.

Are we proposing to explode the groupId? So that 

groupid: org.kie
artifactId: myArtifact

would be 

`org/kie/myartifact/1.0.0.SNAPSHOT` 

?

Anyway any format is fine by me

Reply:
Okay, filed https://issues.redhat.com/browse/DROOLS-6175

Reply:
Thanks guys :+1: